core/View.php

```diff
@@ -458,7 +459,23 @@ public static function singleReport($title, $reportHtml)
     private function shouldPropagateTokenAuthInAjaxRequests()
     {
         $generalConfig = Config::getInstance()->General;
-        return Common::getRequestVar('module', false) == 'Widgetize' || $generalConfig['enable_framed_pages'] == '1';
+        return Common::getRequestVar('module', false) == 'Widgetize' ||
+            $generalConfig['enable_framed_pages'] == '1' ||
+            $this->validTokenAuthInUrl();
+    }
+
+    /**
+     * @param bool $return the token_auth $_GET variable
+     * @return bool|string
+     * @throws Exception
+     */
+    private function validTokenAuthInUrl()
+    {
+        $tokenAuth = Common::getRequestVar('token_auth', '', 'string', $_GET);
+        if ($tokenAuth && $tokenAuth === Piwik::getCurrentUserTokenAuth()) {
+            return $tokenAuth;
+        }
+        return false;
```

Comment:
Guess this can be simplified as there is no need to return the token anymore:

```suggestion
     * @return bool
     * @throws Exception
     */
    private function validTokenAuthInUrl()
    {
        $tokenAuth = Common::getRequestVar('token_auth', '', 'string', $_GET);
        return ($tokenAuth && $tokenAuth === Piwik::getCurrentUserTokenAuth());
```